### ISSUE TITLE
Parsing Cray pointer in Loki

### ISSUE THREAD

**Issue by dhaumont** [ecmwf-ifs/loki#338]
Parsing files containing Cray pointers output this error:

 ```
No specific handler for node type <class 'fparser.two.Fortran2003.Cray_Pointer_Stmt'>

No specific handler for node type <class 'fparser.two.Fortran2003.Cray_Pointer_Decl_List'>
```
The message indicates that the fparser frontend successfully parsed the statements as Cray Pointer statements but the translation into a Loki IR failed because there is no representation for them in Loki.

**Comment by reuterbal** [comment]
Thanks. Could you add a small Fortran reproducer for this?

**Comment by dhaumont** [comment]
@reuterbal Yes, I am working on the reproducer. Coming soon...

**Comment by dhaumont** [comment]
[cray_pointer.tar.gz](https://github.com/user-attachments/files/16066695/cray_pointer.tar.gz)
@reuterbal 
Here is the reproducer

**Comment by reuterbal** [comment]
Thanks. Slightly modified this to a pytest reproducing the issue:
```python
@pytest.mark.parametrize('frontend', available_frontends())
def test_cray_pointers(frontend):
    fcode = """
SUBROUTINE SUBROUTINE_WITH_CRAY_POINTER (KLON,KLEV)
IMPLICIT NONE
INTEGER, INTENT(IN) :: KLON, KLEV
REAL, DIMENSION(KLON,KLEV) :: ZQ
POINTER (IP_ZQ, ZQ)
END SUBROUTINE
    """.strip()
    routine = Subroutine.from_source(fcode, frontend=frontend)
```

For this, the frontends produce the following:

* OMNI: ` line 5: compiler error: compile_statement1: not implemented yet (F-compile.c:747)\n`
* OFP: 
```xml
          <declaration col_begin="0" col_end="20" line_begin="5" line_end="5">
            <cray-pointer-assoc-list__begin addendum="list-begin" rule="541000" />
            <cray-pointer-assoc col_begin="9" col_end="18" line_begin="5" line_end="5" pointee="ZQ" pointer="IP_ZQ" rule="541000" />
            <cray-pointer-assoc-list addendum="list" count="1" rule="541000" />
            <cray-pointer-stmt col_begin="0" col_end="20" eos="&#10;" keyword="POINTER" line_begin="5" line_end="5" rule="540000" />
            <specification-stmt rule="212" />
          </declaration>
```
* fparser: `Cray_Pointer_Stmt('POINTER', Cray_Pointer_Decl_List(',', (Cray_Pointer_Decl(Name('IP_ZQ'), Name('ZQ')),)))`

As a quick fix, I will implement representing the pointer statement as an `Intrinsic` node for OFP and Fparser.

**Comment by dhaumont** [comment]
Thanks Balthasar, the reproducer looks much better like that.
I will be happy with the workaround, please tell me when it will be available.

**Comment by reuterbal** [comment]
A fix has been implemented in aea722fd60fa67d0e34fd9bdc161a566ca615fce and filed as a PR. It may take a little while until this lands on the main branch but feel free to test this in the meantime.
